addons/website_sale/controllers/main.py

```diff
@@ -1215,20 +1226,40 @@ def process_express_checkout(self, billing_address, **kwargs):
             child_partner_id = self._find_child_partner(
                 order_sudo.partner_id.commercial_partner_id.id, billing_address
             )
-            if child_partner_id:
-                order_sudo.partner_invoice_id = child_partner_id
-            else:
-                billing_partner_id = self._create_or_edit_partner(
-                    billing_address,
-                    type='invoice',
-                    parent_id=order_sudo.partner_id.id,
+            order_sudo.partner_invoice_id = child_partner_id or self._create_or_edit_partner(
+                billing_address, type='invoice', parent_id=order_sudo.partner_id.id
+            )
+
+        if shipping_address:
+
+            self._include_country_and_state_in_address(shipping_address)
+
+            if order_sudo.partner_shipping_id.name.endswith(order_sudo.name):
+                # The existing partner was created by `process_express_checkout_delivery_choice`, it
+                # means that the partner is missing information, so we update it.
+                order_sudo.partner_shipping_id = self._create_or_edit_partner(
+                    shipping_address,
+                    edit=True,
+                    type='delivery',
+                    partner_id=order_sudo.partner_shipping_id.id,
+                )
+            elif any(
+                shipping_address[k] != order_sudo.partner_shipping_id[k] for k in shipping_address
+            ):
+                # The sale order's shipping partner's address is different from the one received. If
+                # all the sale order's child partners' address differs from the one received, we
+                # create a new partner. The phone isn't always checked because it isn't sent in
+                # shipping information with Google Pay.
+                child_partner_id = self._find_child_partner(
+                    order_sudo.partner_id.commercial_partner_id.id, shipping_address
+                )
+                order_sudo.partner_shipping_id = child_partner_id or self._create_or_edit_partner(
+                    shipping_address, type='delivery', parent_id=order_sudo.partner_id.id
                 )
-                order_sudo.partner_invoice_id = billing_partner_id
 
-        # In a non-express flow, `sale_last_order_id` would be added in the session before the
-        # payment. As we skip all the steps with the express checkout, `sale_last_order_id` must be
-        # assigned to ensure the right behavior from `shop_payment_confirmation()`.
-        request.session['sale_last_order_id'] = order_sudo.id
```

Comment:
No, same PR #113644

Reply:
Then I don't agree, because reading the `request.session['sale_last_order_id'] = order_sudo.id`, I was like "What? Why?" and I searched for this comment. Sorry @chevalierv but I don't usually work on these flows, and without comment, that's not clear.

Reply:
:red_circle: We moved this line in another PR, which is still in development. We moved it for technical reasons during the development, which are no longer causing trouble I think. 

Although I like the spirit of anticipation, I'd rather keep this diff in this PR, since the other has not even been reviewed once and I still don't know if it will stay like this. (And I agree with @Demesmaeker, the comment was very important and should not be dissociated form this line)